t/after.t

```diff
@@ -0,0 +1,20 @@
+use strict;
+use warnings;
+
+use Test::More;
+use Path::Tiny;
+use Rex::Commands;
+
+$::QUIET = 1;
+
+my $tmpfile = Path::Tiny->tempfile;
+
+task  foo => "asdfsadfasdf" => sub { return "yo" };
+after foo => sub { $tmpfile->remove };
+
+Rex::TaskList->run("foo");
+ok !$tmpfile->exists, "after hook runs even when the ssh connection fails";
+
+
+done_testing;
```

Comment:
I prefer having test plans. That being said, we can merge it like this, but I will clean this up later anyways :)